src/bin/cross-util.rs

```diff
@@ -0,0 +1,193 @@
+//  tag = ${target}(-${version})?
+// our target triple must match `[A-Za-z0-9_-]`
+fn rustembedded_target(tag: &str) -> Option<String> {
+    let is_target_char = |c: char| c == '_' || c.is_ascii_alphanumeric();
+    let mut components = vec![];
+    for component in tag.split('-') {
+        if !component.is_empty() && component.chars().all(is_target_char) {
+            components.push(component)
+        }
+    }
+
+    Some(components.join("-"))
+}
+
+fn get_image_target(image: &str) -> cross::Result<String> {
+    let option = if let Some(stripped) = image.strip_prefix(GHCR_IO) {
+        stripped.split(':').next().map(|t| t.to_string())
+    } else if let Some(tag) = image.strip_prefix(RUST_EMBEDDED) {
+        rustembedded_target(tag)
+    } else if let Some(tag) = image.strip_prefix(DOCKER_IO) {
+        rustembedded_target(tag)
+    } else {
+        None
+    };
+
+    match option {
+        Some(target) => Ok(target),
+        None => eyre::bail!("Cannot get target for image {}", image),
+    }
+}
+
+fn list_images(engine: &Path, verbose: bool) -> cross::Result<()> {
+    get_cross_images(engine, verbose, true)?
+        .iter()
+        .for_each(|line| println!("{}", line));
+
+    Ok(())
+}
+
+fn remove_images(
+    engine: &Path,
+    images: &[String],
+    verbose: bool,
+    force: bool,
+) -> cross::Result<()> {
+    let mut command = Command::new(engine);
+    command.arg("rmi");
+    if force {
+        command.arg("--force");
+    }
+    command.args(images);
+    command.run(verbose)
+}
+
+fn remove_all_images(engine: &Path, verbose: bool, force: bool, local: bool) -> cross::Result<()> {
+    let images = get_cross_images(engine, verbose, local)?;
+    remove_images(engine, &images, verbose, force)
+}
+
+fn remove_target_images(
+    engine: &Path,
+    targets: &[String],
+    verbose: bool,
+    force: bool,
+    local: bool,
+) -> cross::Result<()> {
+    let images = get_cross_images(engine, verbose, local)?;
+    let mut filtered = vec![];
+    for image in images {
+        let target = get_image_target(&image)?;
+        if targets.contains(&target) {
+            filtered.push(image);
+        }
+    }
+    remove_images(engine, &filtered, verbose, force)
+}
+
+pub fn main() -> cross::Result<()> {
+    cross::install_panic_hook()?;
+    let cli = Cli::parse();
+    match &cli.command {
+        Commands::ListImages { verbose, engine } => {
+            let engine = get_container_engine(engine)?;
+            list_images(&engine, *verbose)?;
+        }
+        Commands::RemoveImages {
+            targets,
+            verbose,
+            force,
+            local,
+            engine,
+        } => {
+            let engine = get_container_engine(engine)?;
+            if targets.is_empty() {
+                remove_all_images(&engine, *verbose, *force, *local)?;
+            } else {
+                remove_target_images(&engine, targets, *verbose, *force, *local)?;
+            }
```

Comment:
i think there should be a confirm prompt here, or, a we have a `--execute` flag and standard operation without it is a dry-run